File: benhussein/report/price_history_report/price_history_report.py
"""Price History Report."""

import frappe


def execute(filters=None):
    columns = [
        {"fieldname": "price_date", "label": "Date", "fieldtype": "Date", "width": 100},
        {"fieldname": "metal", "label": "Metal", "fieldtype": "Data", "width": 100},
        {"fieldname": "purity", "label": "Purity", "fieldtype": "Data", "width": 80},
        {"fieldname": "price_usd", "label": "USD", "fieldtype": "Currency", "width": 120},
        {"fieldname": "price_local", "label": "Local", "fieldtype": "Currency", "width": 120},
    ]
    data = []
    return columns, data

File: benhussein/report/price_history_report/price_history_report.js
frappe.query_reports["Price History Report"] = {
  filters: [
    {
      fieldname: "metal",
      label: __("Metal"),
      fieldtype: "Select",
      options: ["", "Gold", "Silver", "Platinum", "Palladium"],
    },
    {
      fieldname: "from_date",
      label: __("From"),
      fieldtype: "Date",
    },
    {
      fieldname: "to_date",
      label: __("To"),
      fieldtype: "Date",
    },
  ],
};
